eth-providers/src/base-provider.ts

```diff
@@ -741,8 +743,38 @@ export abstract class BaseProvider extends AbstractProvider {
 
   // Queries
   getTransaction = (transactionHash: string): Promise<TransactionResponse> => throwNotImplemented('getTransaction');
-  getTransactionReceipt = (transactionHash: string): Promise<TransactionReceipt> =>
-    throwNotImplemented('getTransactionReceipt');
+  getTransactionReceipt = async (transactionHash: string): Promise<TransactionReceipt> => {
+    const tx = await getTxReceiptByHash(transactionHash);
+
+    if (!tx) {
+      return logger.throwError(`transaction hash not found`, Logger.errors.UNKNOWN_ERROR, { transactionHash });
+    }
+
+    // TODO: correct values of these?
+    const confirmations = 0;
+    const effectiveGasPrice = BIGNUMBER_ONE;
+    const byzantium: false = false;
+    const defaultAddress = '0x';
+
+    return {
+      to: tx.to || defaultAddress,
```

Comment:
according to https://infura.io/docs/ethereum/json-rpc/eth-getTransactionReceipt it should be null if no to

>  address of the receiver. Null when the transaction is a contract creation transaction.

Reply:
yeah but this would conflict with the type library we use:
https://github.com/ethers-io/ethers.js/blob/b1458989761c11bf626591706aa4ce98dae2d6a9/packages/abstract-provider/src.ts/index.ts#L106
according to it, `to` and `contractAddress` should both the `string`, also `confirmations`, `effectiveGasPrice`, `byzantium` are required.

Is there a more accurate lib that reflect EVM types? Or should we overwrite `@etherproject` to make the type match infura?

Reply:
maybe we should use the parity frontier lib as our type definition?

Reply:
The infura definition seems to be the same as [eth wiki](https://eth.wiki/json-rpc/API#eth_gettransactionreceipt), but don't know why `@ethersproject` definition is so wired... Opened a discussion there but no one answered
https://github.com/ethers-io/ethers.js/discussions/2270

Reply:
ok. I am ok with the current code until something else complains it needs to be null.
can you write an issue and we can merge this

Reply:
sounds good! Created a new issue #100 to address this